### PR TITLE
INTEGRATE-1429: Add `lock` and `unlock` endpoint definitions to OpenAPI

### PR DESCRIPTION
**Pull request by orwigeg**
[INTEGRATE-1429](https://memsourcegroup.atlassian.net/browse/INTEGRATE-1429)

### CODE REVIEW

**Comment by docstun** [comment]
@orwigeg is this still relevant?

**Comment by orwigeg** [comment]
@docstun - It depends. David said we should wait until the job locking and unlocking feature were available outside of a feature toggle. It looks like it's still in the "beta" column in `feature_toggle.rb`. If we don't have any plans on removing it from there, perhaps we can just close the PR.

**Comment by orwigeg** [comment]
@docstun - I just got word from Sönke that we got the greenlight to release job locking while I was away on vacation. We'll move the ticket to our new TSS board and we can keep this PR open and merge it soon.

**Comment by docstun** [comment]
@orwigeg thanks a lot - happy to hear!

File: paths/jobs/lock_job.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+---
+summary: Lock a job
+description: If you are the job owner, you may lock a job using this API request.
+operationId: job/keys/lock
```
Comment (theSoenke) [inline paths/jobs/lock_job.yaml:4]: I wonder whether this should be `job/lock`?

File: paths/jobs/unlock_job.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+---
+summary: Unlock a job
+description: If you are the job owner, you may unlock a locked job using this API request.
+operationId: job/keys/unlock
```
Comment (theSoenke) [inline paths/jobs/unlock_job.yaml:4]: same as above

File: paths/jobs/lock_job.yaml (hunk 1 above)
Comment (orwigeg) [reply paths/jobs/lock_job.yaml:4]: Hm, good point

Comment (orwigeg) [reply paths/jobs/lock_job.yaml:4]: I'll look into this tomorrow morning

Comment (orwigeg) [reply paths/jobs/lock_job.yaml:4]: I feel like there was a reason we did it this way, but I can't remember what it was

Comment (orwigeg) [reply paths/jobs/lock_job.yaml:4]: I looked into it again this morning and for the life of me I can't remember what our reasoning was, so I'll change it :)

**Review by theSoenke — APPROVED** [review]
:tada:

**Pull request merged**